src/condor_utils/ipv6_hostname.cpp

```diff
@@ -176,20 +176,20 @@ std::string get_fqdn_from_hostname(const std::string& hostname) {
 	std::string ret;
 
 	if (!nodns_enabled()) {
-		addrinfo_iterator ai;
-		int res  = ipv6_getaddrinfo(hostname.c_str(), NULL, ai);
+		addrinfo* info = nullptr;
+		int res  = ipv6_getaddrinfo(hostname.c_str(), NULL, info);
 		if (res) {
 			dprintf(D_HOSTNAME, "ipv6_getaddrinfo() could not look up %s: %s (%d)\n", 
 				hostname.c_str(), gai_strerror(res), res);
 			return ret;
 		}
 
-		addrinfo *info = ai.next();
-		if (info != NULL && info->ai_canonname != NULL) {
+		if (info->ai_canonname != nullptr) {
```

Comment:
This is OK if ipv6_getaddrinfo() guarantees that `info` is non-NULL, but if it it's not NULL, then, unless I'm missing something, line 189 is wrong; we need to free `info` before exiting this function, which (also) means copying `ai_canonname`.

Reply:
According to the map page on macOS for getaddrinfo(), the struct addrinfo* will be non-null on a successful call. ipv6_getaddrinfo() maintains that, and we return on failure just above here. I see the linux man page isn't quite that explicit. I will fix the obvious memory leak and we can discuss the non-null assumption tomorrow.